test/parallel/test-crypto-fips.js

```diff
@@ -0,0 +1,194 @@
+'use strict';
+var common = require('../common');
+var assert = require('assert');
+var spawn = require('child_process').spawn;
+var path = require('path');
+
+if (!common.hasCrypto) {
+  console.log('1..0 # Skipped: missing crypto');
+  return;
+}
+
+const EXIT_FAILURE = -1;
+const FIPS_ENABLED = 1;
+const FIPS_DISABLED = 0;
+const FIPS_ERROR_STRING = 'Error: Cannot set FIPS mode';
+const OPTION_ERROR_STRING = 'bad option';
+const CNF_FIPS_ON = common.fixturesDir + '/openssl_fips_enabled.cnf';
+const CNF_FIPS_OFF = common.fixturesDir + '/openssl_fips_disabled.cnf';
+var num_children_spawned = 0;
+var num_children_ok = 0;
+
+function compiledWithFips() {
+  return process.config.variables.openssl_fips ? true : false;
+}
+
+function addToEnv(newVar, value) {
+  var envCopy = {};
+  for (const e in process.env) {
+    envCopy[e] = process.env[e];
+  }
+  envCopy[newVar] = value;
+  return envCopy;
+}
+
+function getResponse(data)
+{
+  return data.toString().replace('\n', '').replace('>', '').trim();
+}
+
+function childOk(child) {
+  console.log('Child ' +  ++num_children_ok + '/' + num_children_spawned +
+      ' [pid:' + child.pid + '] OK.');
+}
+
+function testHelper(requiresFips, args, expectedOutput, cmd, env) {
+  const child = spawn(process.execPath, args.concat(['-i']), {
+    cwd: path.dirname(process.execPath),
+    env: env
+  });
+
+  console.log('Spawned child [pid:' + child.pid + '] with cmd ' +
+              cmd + ' and args \'' + args + '\'');
+  num_children_spawned++;
+
+  // If using FIPS mode binary, or running a generic test.
+  if (compiledWithFips() || !requiresFips) {
+    child.stdin.setEncoding('utf-8');
+    child.stdout.on('data', function(data) {
+      // Prompt and newline may occur in undefined order.
+      const response = getResponse(data);
```

Comment:
Perhaps it should wait until `end` event to accumulate all data? I'm afraid of the situation where `stdout` could be chunked.